src/rubric_eval/main.py

```diff
@@ -0,0 +1,217 @@
+from .annotators import RubricBrainstormer
+
+CUR_DIR = Path(__file__).parent
+
+__all__ = ["get_rubrics", "get_outputs", "eval_outputs"]
+
+
+def check_df_fields(df, required_fields, optional_fields=set()):
+    allowed_fields = required_fields | optional_fields
+    
+    actual_fields = set(df.columns)
+    
+    # Check if all required fields are present
+    if not required_fields.issubset(actual_fields):
+        missing = required_fields - actual_fields
+        raise RuntimeError(f"Missing required fields: {missing}")
+    
+    # Check if there are any fields not in the allowed set
+    extra_fields = actual_fields - allowed_fields
+    if extra_fields:
+        raise RuntimeError(f"Extra fields found: {extra_fields}")
+    
+
+def cleanup_df_rubrics(df_rubrics):
+    # Drop rows where the "scoring_scales" column is not a dict (because sometimes rubric generator will refuse to generate rubric)
+    mask = df_rubrics["scoring_scales"].apply(lambda x: isinstance(x, dict))
+    return df_rubrics[mask]
+
+
+def list_files_with_prefix(path_prefix, suffix):
+    matching_files = []
+    
+    # Expand path_prefix to full absolute path
+    full_path_prefix = os.path.abspath(path_prefix)
+    
+    # Get the directory part of the full_path_prefix
+    prefix_dir = os.path.abspath(os.path.dirname(full_path_prefix))
+    
+    # Walk through the directory
+    for root, _, files in os.walk(prefix_dir):
+        for file in files:
+            full_path = os.path.join(root, file)
+            if (
+                full_path == os.path.join(prefix_dir, file)
+                and full_path.startswith(full_path_prefix)
+                and full_path.endswith(suffix)
+            ):
+                matching_files.append(full_path)
+    
+    return matching_files
+
+
+def string_to_list(s):
+    try:
+        return ast.literal_eval(s)
+    except (ValueError, SyntaxError):
+        # Handle the case where the string is not a valid literal
+        print(f"Error: '{s}' is not a valid list representation")
+        return None
+
+
+
+def get_rubrics(
+    input_path: str = "instructions.json",
+    output_path: str = "instructions_with_rubrics.json",
+):
+    """
+    input_path: str, path to input file (JSON)
+    The JSON file should contain a list of instructions, each instruction is a dictionary containing the following fields:
+    - prompt
+    - category (optional)
+    - additional_information (optional, free-form text)
+
+    TODO: should we allow more fields in the input JSON file that are just ignored, so that people don't need to manually remove those fields in their dataset JSON file before calling this function?
+    TODO: should we support other serialization formats like pickle, Parquet, HDF5, etc.?
+    TODO: should we support another Python function calling this function, thus needing to keep both input and output in memory (for perf reason) instead of being serialized to disk?
```

Comment:
see above: if you use the function I gave you then we can also move around dataframes in memory. Basically i think all functions should be able to have in memory objects as input & output. At the start and end of the function you can load or save with the function from alpaca_eval but then assume that everything is in memory